dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -268,34 +256,40 @@ private void checkContentPermission(final UrlMapContext context, final Contentle
                 contentlet, PermissionLevel.READ.getType(), context.getUser(), context.getMode().respectAnonPerms);
 
         if (!havePermission) {
-            throw new DotSecurityException(String.format("User dont have permission in content: %s", contentlet.getName()));
+            throw new DotSecurityException(String.format("User does not have permission in content: %s", contentlet.getName()));
         }
     }
 
     private String buildContentQuery(
             final Matches matches,
-            final Structure structure,
-            final Field hostField,
+            final ContentType structure,
             final UrlMapContext context) {
 
         final StringBuilder query = new StringBuilder();
 
-        query.append("+structureName:").append(structure.getVelocityVarName())
-             .append(" +deleted:false ");
-
-        if (context.getMode() == PageMode.PREVIEW_MODE || context.getMode() == PageMode.EDIT_MODE) {
-            query.append("+working:true ");
+        query.append("+contentType:")
+            .append(structure.variable())
+            .append(" +deleted:false ")
+            .append(" +(conhost:")
+                .append(context.getHost().getIdentifier())
+                .append(" OR conhost:")
+                .append(Host.SYSTEM_HOST)
+            .append(")");
+        
+
+        if (context.getMode().showLive) {
+            query.append(" +live:true ");
         } else {
-            query.append("+live:true ");
+            query.append(" +working:true ");
         }
 
-        if (null != hostField && context.getHost() != null) {
-            query.append(this.getHostFilter(context.getHost()));
-        }
+
 
         query.append(" ");
-        query.append(this.buildFields(structure, matches))
-             .append(" +languageId:").append(context.getLanguageId());
+        query.append(this.buildFields(structure, matches));
```

Comment:
Codacy found an issue: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796110)